src/BatchResult.php

```diff
@@ -117,29 +109,21 @@ public function getExceptions()
     /**
      * {@inheritDoc}
      */
-    public function getExceptionFor(RequestInterface $request)
-    {
-        try {
-            return $this->exceptions[$request];
-        } catch (\UnexpectedValueException $e) {
-            throw new UnexpectedValueException('Request not found', $e->getCode(), $e);
-        }
-    }
-
-    /**
-     * {@inheritDoc}
-     */
-    public function hasExceptions()
+    public function isFailed(RequestInterface $request)
     {
-        return $this->exceptions->count() > 0;
+        return $this->exceptions->contains($request);
     }
 
     /**
      * {@inheritDoc}
      */
-    public function hasExceptionFor(RequestInterface $request)
+    public function getExceptionFor(RequestInterface $request)
     {
-        return $this->exceptions->contains($request);
+        try {
+            return $this->exceptions[$request];
+        } catch (\UnexpectedValueException $e) {
+            throw new UnexpectedValueException('Request not found', $e->getCode(), $e);
```

Comment:
I think I did this rethrow thing in order to hide the original Object not found message. But the question is: If it is the same, why isn't it used from the root namespace as well?